torch/csrc/api/include/torch/nn/functional/distance.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include <torch/nn/options/distance.h>
+
+namespace torch {
+namespace nn {
+namespace functional {
+
+inline Tensor cosine_similarity(
+    const Tensor& x1,
+    const Tensor& x2,
+    const CosineSimilarityOptions& options) {
+  return torch::cosine_similarity(
+      x1,
+      x2,
+      options.dim_,
```

Comment:
We should use `options.dim()` instead of `options.dim_`, because `options.dim_` will be made private in the near future (https://github.com/pytorch/pytorch/pull/26419) (similarly for other options args as well).